### ISSUE TITLE
the QC shiny app should be deleted from oceglider

### ISSUE THREAD

**Issue by dankelley** [dankelley/oceanglider#124]
In a meeting on 2024-09-11, the co-authors decided that the QC shiny app ought to be removed from the oceglider package.  I will do that, and make a new repository for the app.  The reasons include

1. The app is not in current use by the co-authors or their collaborators.
2. The app was designed for use with a data format that has evolved somewhat over the years, so that it might not even work if tried on modern data.
3. We are in a trimming phase, in which oceglider is shedding temporary code and focussing on core capabilities.

**Comment by dankelley** [comment]
The deletion was done in commit d7e26ae1732a6afe5725ef81dba3ff01294de97f of branch "develop".

**Comment by dankelley** [comment]
The code is now present in commit  https://github.com/dankelley/ocean-glider-QC/commit/0fdb32de9610f8b5799c184c060d1dd352edc8ef of the "main" branch of repo https://github.com/dankelley/ocean-glider-QC

**Comment by clayton33** [comment]
Since this shiny app was deleted, should `shiny` be removed from `Imports` in the `DESCRIPTION` file ?

**Comment by dankelley** [comment]
Good catch!  I see that `read.glider.seaexplorer.delayed()` has a parameter called `progressBar` that is allowed to be TRUE, FALSE, or "shiny".  In the last condition, the progress bar is shown using the shiny tool.  This comes up in code like the following
```R
        if (is.logical(progressBar) && progressBar) {
            setTxtProgressBar(pb, i)
        } else if (is.character(progressBar) && progressBar == "shiny") {
            shiny::incProgress(1 / nfiles)
        }
```
and I think that second part can be removed.  Unless you or another person objects, I'll do that tomorrow morning.

I will also likely reformat the code to the way that neovim does it (the same as RStudio, I think) just because that might make it easier for users to read through.  But I'd do that in a commit of its own, with a good commit note, so that folks will understand the reason for a change that affects the 'git diff' but does not affect how anything works.

Thanks!

**Comment by dankelley** [comment]
A checklist (made by `rg -n shiny:: R|grep -v docs`):

* [ ] `R/seaexplorer_realtime.R:45:#'     then [shiny::setProgress()] and [shiny::incProgress()] will be used,`
* [ ] `R/seaexplorer_realtime.R:47:#'     was made within the context of a call to [shiny::withProgress()].`
* [ ] `R/seaexplorer_realtime.R:187:        shiny::setProgress(0, paste("reading", nfiles, "files"))`
* [ ] `R/seaexplorer_realtime.R:194:            shiny::incProgress(1 / nfiles)`
* [ ] `R/seaexplorer_realtime.R:211:        shiny::setProgress(0, paste("reading", nfiles, "files"))`
* [ ] `R/seaexplorer_realtime.R:218:            shiny::incProgress(1 / nfiles)`
* [ ] `R/seaexplorer_delayed.R:94:#'     then [shiny::setProgress()] and [shiny::incProgress()] will be used,`
* [ ] `R/seaexplorer_delayed.R:96:#'     was made within the context of a call to [shiny::withProgress()].`
* [ ] `R/seaexplorer_delayed.R:183:        shiny::setProgress(0, paste("reading", nfiles, "files"))`
* [ ] `R/seaexplorer_delayed.R:189:            shiny::incProgress(1 / nfiles)`
* [ ] `R/seaexplorer_delayed.R:375:            shiny::setProgress(0, paste("handling NA values for", n, "variables"))`
* [ ] `R/seaexplorer_delayed.R:383:                    shiny::incProgress(1 / n, paste("handling NA for", var))`
* [ ] `R/seaexplorer_delayed.R:404:            shiny::setProgress(0, "removing duplicated items")`
* [ ] `R/seaexplorer_delayed.R:409:            shiny::setProgress(0, "calculating salinity")`
* [ ] `R/seaexplorer_delayed.R:419:        shiny::setProgress(0, "creating units")`

**Comment by clayton33** [comment]
ah ! OK. well if you think the shiny progress bar should stay, then keep it. I hadn't realized it was there. The only reason I caught this in the first place is because I didn't have `shiny` installed and had remembered that you removed the app, so I assumed that was the only place it was used.

**Comment by dankelley** [comment]
The docs (Roxygen2 format) are as follows right now.  I think this may be okay.  Do you think it needs more clarity?

```R
#' @param progressBar either a logical or character value that controls
#'     whether/how to indicate the progress made in reading and interpreting
#'     the data.  This can be useful, since the work can be slow.  If `progressBar`
#'     is a logical value, then it indicates whether to show textual progress
#'     with [txtProgressBar()].  If `progressBar` is the character value `"shiny"`,
#'     then [shiny::setProgress()] and [shiny::incProgress()] will be used,
#'     on the assumption that the call to [read.glider.seaexplorer.realtime()]
#'     was made within the context of a call to [shiny::withProgress()].
#'     The default is to use the value returned by [interactive()], i.e.
#'     to use a textual progress indicator, but only in interactive mode.
```

**Comment by dankelley** [comment]
Maybe see https://lubridate.tidyverse.org/reference/round_date.html if you want to do `ceiling`.

**Comment by clayton33** [comment]
I don't really understand this part 

```
#'     on the assumption that the call to [read.glider.seaexplorer.realtime()]
#'     was made within the context of a call to [shiny::withProgress()]
```
but maybe it's because I don't have familiarity with the shiny progress bar.

**Comment by dankelley** [comment]
That *is* confusing.  I think the idea is that a person can wrap shiny calls in `shiny::withProgress()` and that tells the server to monitor progress, showing some kind of notation that it is "thinking".  As I recall, there are ways to make it bounce or jiggle, or have hands of a clock or whatnot.  But, again as I recall, I grew to dislike that because it seemed to itself slow things down.

I'm not actually sure this matters.  I think folks are more likely to download files independently of any analysis app they might write. Also, if the data are in netcdf format then the download and reading ought to be quick, so no need for a progress bar.  (I think a progress bar only really helps for things that take maybe over 30 seconds, when the user starts to wonder if something is wrong.)

I now propose -- and sorry to go back and forth on this -- that I just remove the option for these functions to create `shiny` progress bars.  If the user wants to see console-type shiny bars, great.  I suspect users will be running Rstudio (or other) locally anyway, so they can see the console progress bars.  If the R is running remotely on a machine that serves up shiny output, that's a different story.  But running stuff on remote machines is not that sensible for interactive work, I think.

OK, I'm rambling.  @clayton33 are you OK if I just remove the shiny progress bar stuff?  I don't want to revisit this tomorrow and you seem to be online now and maybe we can just decide.

**Comment by clayton33** [comment]
+1 vote from me to remove the shiny progress bar stuff.

**Comment by dankelley** [comment]
I'm closing this because it was handled a while back. When I do
```
git grep -i shiny | grep -v docs
```
I get no hits, and this proves that the shiny app has been removed.
